client/api_provider.go

```diff
@@ -41,16 +40,20 @@ const (
 
 // Fetches configuration directly from Lekko Backend APIs.
 func ConnectAPIProvider(ctx context.Context, apiKey string, rk *RepositoryKey, opts ...ProviderOption) (Provider, error) {
-	if rk == nil {
-		return nil, fmt.Errorf("no repository key provided")
+	cfg := &providerConfig{}
+	for _, opt := range opts {
```

Comment:
maybe apply opts after built-in options to allow users to override default behavior?

Reply:
same in other places

Reply:
the built-in options below this line are mandatory, which is why they come after. same reason in the other spots. i can add a comment to make that clear.